Feature/expose new social meta

jomurgel opened the pull request:
This PR exposes additional data into the REST endpoint.  Those data points are:

- `yoast_wpseo_facebook_title` aka `yoast_wpseo_opengraph-title`
- `yoast_wpseo_facebook_description` aka `yoast_wpseo_opengraph-description`
- `yoast_wpseo_facebook_type`
- `yoast_wpseo_facebook_image` aka `yoast_wpseo_opengraph-image`
- `yoast_wpseo_twitter_title`
- `yoast_wpseo_twitter_description`
- `yoast_wpseo_twitter_image`
- `yoast_wpseo_social_url`
- `yoast_wpseo_website_name`
- `yoast_wpseo_company_or_person`
- `yoast_wpseo_person_name`
- `yoast_wpseo_company_name`
- `yoast_wpseo_company_logo`
- `yoast_wpseo_website_name`
- `yoast_wpseo_social_defaults`

The defaults include the following:
- `facebook_site`
- `instagram_url`
- `linkedin_url`
- `myspace_url`
- `og_default_image`
- `og_frontpage_title`
- `og_frontpage_desc`
- `og_frontpage_image`
- `opengraph`
- `pinterest_url`
- `pinterestverify`
- `plus-publisher`
- `twitter`
- `twitter_site`,
- `twitter_card_type`
- `youtube_url`
- `google_plus_url`
- `fbadminapp`

This is on top of the title, meta desc, and the canonical URL.

I've also updated the readme to reflect this change.

jomurgel committed:
exposes new data in yoast rest api endpoint — cleanup

jomurgel committed:
updates readme with new data

phatsk commented on plugin.php:
```diff
@@ -119,17 +126,37 @@ function wp_api_encode_yoast( $p, $field_name, $request ) {
+			'yoast_wpseo_twitter_description'  => get_post_meta( $p['id'], '_yoast_wpseo_twitter-description', true ),
+			'yoast_wpseo_twitter_image'        => get_post_meta( $p['id'], '_yoast_wpseo_twitter-image', true ),
+			'yoast_wpseo_social_url'           => get_permalink( $p['id'] ),
+			'yoast_wpseo_website_name'         => $yoast_titles['website_name'],
```
If we're using PHP7, I would suggest using the null coalesce `??` operator on these lines:
```php
'yoast_wpseo_website_name' => $yoast_titles['website_name'] ?? '',
```
or pre-7 something like
```php
'yoast_wpseo_website_name' => isset( $yoast_titles['website_name'] ) ? $yoast_titles['website_name'] : '',
```
OR, prep the array with defaults:
```php
$yoast_titles = wp_parse_args( get_option( 'wpseo_titles', [] ), [
    'website_name' => '',
    // etc
] );
```

phatsk commented on plugin.php:
```diff
@@ -170,9 +198,11 @@ private function wp_api_encode_taxonomy() {
-		query_posts( array(
-			'cat' => $category['id'],
-		) );
+		query_posts(
```
This could all be on a single line:
```php
query_posts( [ 'cat' => $category['id'] ] );
```

phatsk commented on plugin.php:
```diff
@@ -184,9 +214,11 @@ function wp_api_encode_yoast_category( $category ) {
 	}
 
 	function wp_api_encode_yoast_tag( $tag ) {
-		query_posts( array(
```
Same note as above about being on a single line.

phatsk reviewed (approved):
All-in-all looks good, maybe could use some minor adjustments like ensuring the options are present before use (or defaulting to a value).

jomurgel committed:
query_posts array cleanup

jomurgel committed:
outputs empty strings if data doesn't exist — file cleanup

jomurgel committed:
lints file

jomurgel commented:
Made a couple of updates here. Primarily adding null-coalesce operators to some of the data. Confirmed we'll need at least PHP version 7 in order to run our app, so this is ok.

Also renamed the object key on line 185, to match the same object on line 159. Should have been `yoast_wpseo_social_defaults`.

Addressed the code notes above. 👍 

Also updated some things to pass linting.

phatsk commented on plugin.php:
```diff
@@ -119,17 +123,37 @@ function wp_api_encode_yoast( $p, $field_name, $request ) {
+			'yoast_wpseo_company_name'         => $yoast_titles['company_name'] ?? '',
+			'yoast_wpseo_company_logo'         => $yoast_titles['company_logo'] ?? '',
+			'yoast_wpseo_website_name'         => $yoast_titles['website_name'] ?? '',
+			'yoast_wpseo_social_defaults'      => get_option( 'wpseo_social' ) ?? '',
```
Sorry, this line doesn't need the null coalesce, although if you want it to default to empty _string_ you can use:
```php
get_option( 'wpseo_social', '' )
```
to specify `''` as the default value.

jomurgel replied:
LEARNING!

phatsk commented on plugin.php:
```diff
@@ -153,8 +177,9 @@ private function wp_api_encode_taxonomy() {
-			'yoast_wpseo_metadesc' => $wpseo_frontend->metadesc( false ),
+			'yoast_wpseo_title'           => $wpseo_frontend->get_taxonomy_title(),
+			'yoast_wpseo_metadesc'        => $wpseo_frontend->metadesc( false ),
+			'yoast_wpseo_social_defaults' => get_option( 'wpseo_social' ) ?? '',
```
Same note about null coalesce and `get_option`

phatsk reviewed (approved):
Looks good! Just two adjustments around `get_option` and `??`

jomurgel committed:
remove null coalesce from get_options

jomurgel commented:
Removed those two instances of null coalesce in the `get_options` calls.

Also removed a duplicate of `yoast_wpseo_website_name` I didn't catch.

jomurgel committed:
removes duplicate yoast_wpseo_website_name
